pyro/distributions/omt_mvn.py

```diff
@@ -0,0 +1,112 @@
+from __future__ import absolute_import, division, print_function
+
+import torch
+from torch.autograd import Function
+from torch.autograd.function import once_differentiable
+from torch.distributions import constraints
+from torch.distributions.utils import lazy_property
+
+from pyro.distributions.torch.multivariate_normal import MultivariateNormal
+
+
+class OMTMultivariateNormal(MultivariateNormal):
+    """Multivariate normal (Gaussian) distribution with OMT gradients w.r.t. both
+    parameters. Note the gradient computation w.r.t. the Cholesky factor has cost
+    O(D^3), although the resulting gradient variance is generally expected to be lower.
+
+    A distribution over vectors in which all the elements have a joint Gaussian
+    density.
+
+    :param torch.autograd.Variable loc: Mean.
+    :param torch.autograd.Variable scale_tril: Cholesky of Covariance matrix.
+    """
+    params = {"loc": constraints.real, "scale_tril": constraints.lower_triangular}
+    support = constraints.real
+    has_rsample = True
+    reparameterized = True
+
+    def __init__(self, loc, scale_tril):
+        self.loc = loc
+        self.scale_tril = scale_tril
+        assert(len(loc.size()) == 1), "OMTMultivariateNormal loc must be 1-dimensional"
+        assert(len(scale_tril.size()) == 2), "OMTMultivariateNormal scale_tril must be 2-dimensional"
+
+    @property
+    def variance(self):
+        return self.covariance_matrix.diag()
+
+    @lazy_property
```

Comment:
Remove this; the computation was already needed to create the base class.

Reply:
thanks! done